### PR TITLE
3606 sync buffer integration error show when item was ignored

### LINKED ISSUE #3606: Sync buffer intergration_error: show when item was ignored
## Is your feature request related to a problem? Please describe 👀

For example, in the MasterListNameJoinTranslation, an item is ignored if the name_id is empty:
```rust
PullTranslateResult::Ignored("Missing name_id".to_string()
```

However, the "intergration_error" says: "Translator for record not found". Think this could say: "Ignored: Missing name_id" instead?

I.e. put the Ignored message into the integration_error.

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #3606

# 👩🏻‍💻 What does this PR do?
Displays the ignored message in sync buffer since it's only currently being logged in the logs which might not be enabled for a lot of devs XD. 

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Have some records that aren't supported by OMS
- [ ] Initialise
- [ ] Go to sync_buffer table and see `Ignored` integration errors
- [ ] Or make sure tests are still alright XD

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: server/service/src/sync/test/test_data/store.rs [hunk 1]
```diff
@@ -114,7 +114,7 @@ const STORE_2: (&str, &str) = (
 fn store_2() -> TestSyncIncomingRecord {
     TestSyncIncomingRecord {
         translated_record: PullTranslateResult::Ignored(
-            "Ignoring not implemented system names".to_string(),
+            "System names not implemented".to_string(),
```
Comment (roxy-dao) [inline server/service/src/sync/test/test_data/store.rs:117]: Ignored messages are already saying `Ignored:` so removing it from here

File: server/service/src/sync/translation_and_integration.rs [hunk 2]
```diff
@@ -107,38 +95,66 @@ impl<'a> TranslationAndIntegration<'a> {
...
-                    result.insert_error(&sync_record.table_name);
-                    warn!(
-                        "{:?} {:?} {:?}",
-                        error, sync_record.record_id, sync_record.table_name
-                    );
-                    // Next sync_record
-                    continue;
-                }
-            };
+            if ignored {
```
Comment (roxy-dao) [inline server/service/src/sync/translation_and_integration.rs:140]: Want to continue onto next record if current record is ignored

File: server/service/src/sync/translation_and_integration.rs [hunk 3]
```diff
@@ -107,38 +95,66 @@ impl<'a> TranslationAndIntegration<'a> {
...
+                        )?;
+                        result.insert_error(&sync_record.table_name);
+
+                        debug!(
+                            "Ignored record: {:?} {:?} {:?}",
+                            ignore_message, sync_record.record_id, sync_record.table_name
+                        );
+                        continue;
+                    }
+                    PullTranslateResult::NotMatched => {}
```
Comment (roxy-dao) [inline server/service/src/sync/translation_and_integration.rs:136]: I feel like we want a message for `NotMatched` as well... currently it just says `Translator for record not found`.

File: server/service/src/sync/translations/invoice.rs [hunk 4]
```diff
@@ -341,8 +351,6 @@ impl SyncTranslation for InvoiceTranslation {
             return Err(anyhow::anyhow!("Invoice not found"));
         };
 
-        // log::info!("Translating invoice row: {:#?}", invoice_row);
```
Comment (roxy-dao) [inline server/service/src/sync/translations/invoice.rs:344]: Not sure why this was here, but seemed okay to remove

File: server/service/src/sync/translations/invoice.rs [hunk 5]
```diff
@@ -433,12 +453,6 @@ impl SyncTranslation for InvoiceTranslation {
 
         let json_record = serde_json::to_value(legacy_row)?;
 
-        // log::info!(
```
Comment (roxy-dao) [inline server/service/src/sync/translations/invoice.rs:436]: Not sure why this was here, but is commented out so thought it was okay to remove

File: server/service/src/sync/translations/special/name_to_name_store_join.rs [hunk 6]
```diff
@@ -50,12 +50,6 @@ impl SyncTranslation for NameToNameStoreJoinTranslation {
         let name_store_joins = NameStoreJoinRepository::new(connection)
             .query_by_filter(NameStoreJoinFilter::new().name_id(EqualFilter::equal_to(&data.ID)))?;
 
-        if name_store_joins.is_empty() {
```
Comment (roxy-dao) [inline server/service/src/sync/translations/special/name_to_name_store_join.rs:53]: This was always empty... so couldn't get the names/name store joins anymore........

File: server/service/src/sync/translations/stocktake.rs [hunk 7]
```diff
@@ -48,9 +48,6 @@ pub struct LegacyStocktakeRow {
     #[serde(deserialize_with = "empty_str_as_option_string")]
     pub inventory_reduction_id: Option<String>,
 
-    // Ignore invad_reductions_ID for V1
```
Comment (roxy-dao) [inline server/service/src/sync/translations/stocktake.rs:51]: invad_reductions_ID is already being translated above.

**Review by clemens-msupply — APPROVED** [review]
Looks good, thank you!

**Pull request merged**